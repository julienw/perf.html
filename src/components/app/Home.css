/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

.collapsing-timeline {
  flex: 1;
}

.home {
  width: 100%;
  min-height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;
  line-height: 1.5;
}

.homeDrop {
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  display: flex;
  align-items: stretch;
  justify-content: center;
  background-color: rgba(255, 255, 255, 0.9);
  padding: 1em;
  box-sizing: border-box;
  opacity: 0;
  pointer-events: none;
  transition: opacity 200ms;
}

.homeDropMessage {
  width: 100%;
  border: 2px dashed #888;
  font-size: 30px;
  color: #666;
  text-align: center;
  border-radius: 10px;
  display: flex;
  align-items: center;
  justify-content: center;
}

.homeDrop.dragging {
  opacity: 1;
}

.homeSpecialMessage {
  padding: 8px 16px;
  background-color: #798FC8;
  border: 1px solid #000;
  border-radius: 3px;
  color: #fff;
  margin: 17px 0;
}

.homeSection {
  position: relative;
  width: 75%;
  max-width: 1200px;
  box-sizing: border-box;
  border: 1px solid #CCC;
  border-radius: 3px;
  padding: 8em;
  font-size: 130%;
  box-shadow: 0 5px 25px #0b1f50;
  background-color: #fff;
  overflow: hidden;
}

.homeSection kbd {
  display: inline-block;
  border: 1px solid #CCC;
  border-radius: 0.2em;
  margin: 0 0.2em;
  padding: 0.0em 0.5em;
  background-color: #F6F6F6;
  box-shadow: 0.1em 0.1em 0 #BBB;
}

.homeTitle {
  margin-top: 0;
  margin-bottom: 0.5em;
  padding-bottom: 0.5em;
  border-bottom: 1px solid #DDD;
}

.homeTitleSubtext {
  font-weight: normal;
}

.homeTitleGithubIcon {
  float: right;
}

.homeTitleGithubIcon svg {
  vertical-align: middle;
}

.homeSectionItems {
  margin: 1em 0;
}

.homeSectionScreenshot {
  width: 390px;
  max-width: 100%;
  height: auto;
  border: 1px solid #d1d1d1;
  border-radius: 0.4em;
  box-shadow: 0 2px 4px #ddd;
}

.homeSectionInstallButton {
  display: inline-block;
  padding: 0.7em;
  border: 2px solid #557de0;
  color: #557de0;
  text-decoration: none;
  transition: background-color 300ms;
  font-weight: bold;
  font-size: 16px;
}

.homeSectionInstallButton:hover {
  color: #fff;
  background-color: #557de0;
}

.homeSectionPlus {
  font-size: 26px;
  line-height: 0;
  position: relative;
  top: 2px;
  padding-right: 12px;
  padding-left: 5px;
}

.homeSectionPerfScreenshot {
  width: 1346px;
  max-width: 90%;
  height: auto;
  margin-top: 15px;
  margin-bottom: -40px;
}

.homeInstructions {
  display: flex;
  margin-top: 30px;
}

.homeInstructionsLeft,
.homeInstructionsRight {
  width: 50%;
}

.homeInstructionsRight {
  padding-left: 30px;
}

.homeInstructionsTransitionGroup {
  position: relative;
}

.homeTransition-enter {
  opacity: 0.1;
  transform: translateX(100px);
}

.homeTransition-enter.homeTransition-enter-active {
  opacity: 1;
  transform: translateX(0);
  transition: opacity 300ms ease-out, transform 300ms ease-out;
}

@media (max-width: 1300px) {
  .homeSection {
    padding: 5em;
  }
}

@media (max-width: 1100px) {
  .homeSection {
    padding: 3em;
    max-width: 80%;
  }
  .homeInstructions {
    display: block;
  }
  .homeInstructionsLeft,
  .homeInstructionsRight {
    width: 100%;
  }
  .homeInstructionsLeft {
    text-align: center;
  }
  .homeInstructionsRight {
    padding-left: 0;
  }
}
