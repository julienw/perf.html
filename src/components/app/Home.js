/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

// @flow

import * as React from 'react';
import { connect } from 'react-redux';
import classNames from 'classnames';
import AddonScreenshot from '../../../res/gecko-profiler-screenshot-2016-12-06.png';
import PerfScreenshot from '../../../res/perf-screenshot-2017-09-08.jpg';
import { retrieveProfileFromFile } from '../../actions/receive-profile';
import { CSSTransition, TransitionGroup } from 'react-transition-group';
import FooterLinks from './FooterLinks';
require('./Home.css');

const ADDON_URL =
  'https://raw.githubusercontent.com/devtools-html/Gecko-Profiler-Addon/master/gecko_profiler.xpi';
const LEGACY_ADDON_URL =
  'https://raw.githubusercontent.com/devtools-html/Gecko-Profiler-Addon/master/gecko_profiler_legacy.xpi';

const InstallButton = ({
  name,
  xpiUrl,
  children,
  className,
}: InstallButtonProps) => {
  return (
    <a
      href={xpiUrl}
      className={className}
      onClick={e => {
        if (window.InstallTrigger) {
          window.InstallTrigger.install({ [name]: xpiUrl });
        }
        e.preventDefault();
      }}
    >
      {children}
    </a>
  );
};

type InstallButtonProps = {
  name: string,
  xpiUrl: string,
  children?: React.Node,
  className?: string,
};

type UploadButtonProps = {
  retrieveProfileFromFile: File => void,
};

class UploadButton extends React.PureComponent<UploadButtonProps> {
  _input: HTMLInputElement | null;

  constructor(props: UploadButtonProps) {
    super(props);
    (this: any)._upload = this._upload.bind(this);
  }

  render() {
    return (
      <div>
        <input
          type="file"
          ref={input => {
            this._input = input;
          }}
          onChange={this._upload}
        />
      </div>
    );
  }

  _upload() {
    if (this._input) {
      this.props.retrieveProfileFromFile(this._input.files[0]);
    }
  }
}

function InstructionTransition(props: {}) {
  return (
    <CSSTransition
      {...props}
      classNames="homeTransition"
      timeout={300}
      exit={false}
    />
  );
}

/**
 * Provide a global function for the add-on to to notify this component that it has
 * been installed.
 */
let homeInstance;
window.geckoProfilerAddonInstalled = function() {
  if (homeInstance) {
    // Forces the Home component to re-evaluate window.isGeckoProfilerAddonInstalled.
    homeInstance.addonInstalled();
  }
};

type HomeProps = {
  specialMessage?: string,
  retrieveProfileFromFile: File => void,
};

type HomeState = {
  isDragging: boolean,
  isAddonInstalled: boolean,
};

class Home extends React.PureComponent<HomeProps, HomeState> {
  _supportsWebExtensionAPI: boolean;
  _isFirefox: boolean;

  constructor(props: HomeProps) {
    super(props);
    (this: any)._startDragging = this._startDragging.bind(this);
    (this: any)._stopDragging = this._stopDragging.bind(this);
    (this: any)._handleProfileDrop = this._handleProfileDrop.bind(this);
    this.state = {
      isDragging: false,
      isAddonInstalled: Boolean(window.isGeckoProfilerAddonInstalled),
    };

    this._supportsWebExtensionAPI = _supportsWebExtensionAPI();
    this._isFirefox = _isFirefox();
  }

  addonInstalled() {
    this.setState({ isAddonInstalled: true });
  }

  componentDidMount() {
    // Prevent dropping files on the document.
    // Help Flow infer the correct type signature for document.addEventListener.
    document.addEventListener(('drag': string), _preventDefault, false);
    document.addEventListener(('dragover': string), _preventDefault, false);
    document.addEventListener(('drop': string), _preventDefault, false);
    // Let the Gecko Profiler Add-on let the home-page know when it's been installed.
    homeInstance = this;
  }

  componentWillUnmount() {
    document.removeEventListener(('drag': string), _preventDefault, false);
    document.removeEventListener(('dragover': string), _preventDefault, false);
    document.removeEventListener(('drop': string), _preventDefault, false);
  }

  _startDragging(event: Event) {
    event.preventDefault();
    this.setState({ isDragging: true });
  }

  _stopDragging(event: Event) {
    event.preventDefault();
    this.setState({ isDragging: false });
  }

  _handleProfileDrop(event: DragEvent) {
    event.preventDefault();
    if (!event.dataTransfer) {
      return;
    }

    const { files } = event.dataTransfer;
    if (files.length > 0) {
      this.props.retrieveProfileFromFile(files[0]);
    }
  }

  _renderInstructions() {
    const { isAddonInstalled } = this.state;
    if (isAddonInstalled) {
      return this._renderRecordInstructions();
    }
    if (this._supportsWebExtensionAPI) {
      return this._renderInstallInstructions();
    }
    if (this._isFirefox) {
      return this._renderLegacyInstructions();
    }
    return this._renderOtherBrowserInstructions();
  }

  _renderInstallInstructions() {
    return (
      <InstructionTransition key={0}>
        <div className="homeInstructions">
          <div className="homeInstructionsLeft">
            <div style={{ textAlign: 'center' }}>
              <img className="homeSectionScreenshot" src={PerfScreenshot} />
            </div>
          </div>
          <div className="homeInstructionsRight">
            <h2>Getting started</h2>
            <p>
              Install the Gecko Profiler Add-on to start recording a performance
              profile in Firefox, then analyze it and share it with perf.html.
            </p>
            <InstallButton
              name="Gecko Profiler"
              className="homeSectionInstallButton"
              xpiUrl={ADDON_URL}
            >
              <span className="homeSectionPlus">+</span>
              Install add-on
            </InstallButton>
            <p>
              You can also analyze a local profile by either dragging and
              dropping it here or selecting it using the button below.
            </p>
            <UploadButton {...this.props} />
          </div>
        </div>
      </InstructionTransition>
    );
  }

  _renderRecordInstructions() {
    return (
      <InstructionTransition key={1}>
        <div className="homeInstructions">
          <div className="homeInstructionsLeft">
            <p>
              <img className="homeSectionScreenshot" src={AddonScreenshot} />
            </p>
          </div>
          <div className="homeInstructionsRight">
            <h2>Recording profiles</h2>
            <p>
              To start profiling, click on the profiling button, or use the
              keyboard shortcuts. The icon is blue when a profile is recording.
              Hit
              <kbd>Capture Profile</kbd> to load the data into perf.html.
            </p>
            {this._renderShortcuts()}
            <p>
              You can also analyze a local profile by either dragging and
              dropping it here or selecting it using the button below.
            </p>
            <UploadButton {...this.props} />
          </div>
        </div>
      </InstructionTransition>
    );
  }

  _renderLegacyInstructions() {
    return (
      <InstructionTransition key={2}>
        <div className="homeInstructions">
          <div className="homeInstructionsLeft">
            <div style={{ textAlign: 'center' }}>
              <img className="homeSectionScreenshot" src={PerfScreenshot} />
            </div>
          </div>
          <div className="homeInstructionsRight">
            <h2>Recording profiles</h2>
            <p>
              To start recording a performance profile in Firefox, first install
              the{' '}
              <InstallButton name="Gecko Profiler" xpiUrl={LEGACY_ADDON_URL}>
                Gecko Profiler Add-on
              </InstallButton>. Then use the button added to the browser, or use
              the following shortcuts to record a profile. The buttonâ€™s icon is
              blue when a profile is recording. Hit <kbd>Capture Profile</kbd>{' '}
              to load the data into perf.html.
            </p>
            {this._renderShortcuts()}
            <p>
              You can also analyze a local profile by either dragging and
              dropping it here or selecting it using the button below.
            </p>
            <UploadButton {...this.props} />
          </div>
        </div>
      </InstructionTransition>
    );
  }

  _renderOtherBrowserInstructions() {
    return (
      <InstructionTransition key={0}>
        <div className="homeInstructions" key={0}>
          <div className="homeInstructionsLeft">
            <div style={{ textAlign: 'center' }}>
              <img className="homeSectionScreenshot" src={PerfScreenshot} />
            </div>
          </div>
          <div className="homeInstructionsRight">
            <h2>How to view and record profiles</h2>
            <p>
              Recording performance profiles requires{' '}
              <a href="https://www.mozilla.org/en-US/firefox/new/">Firefox</a>.
              However, existing profiles can be viewed in any modern browser. To
              view a profile, either follow a link to a public profile, drag a
              saved local profile onto this screen or select it using the button
              below.
            </p>
            <UploadButton {...this.props} />
          </div>
        </div>
      </InstructionTransition>
    );
  }

  _renderShortcuts() {
    return (
      <p>
        <div>
          <p>
            <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>1</kbd> Stop and start
            profiling
          </p>
          <p>
            <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>2</kbd> Capture and load
            profile
          </p>
        </div>
      </p>
    );
  }

  render() {
    const { isDragging } = this.state;
    const { specialMessage } = this.props;
    return (
      <div
        className="home"
        onDragEnter={this._startDragging}
        onDragExit={this._stopDragging}
        onDrop={this._handleProfileDrop}
      >
        <section className="homeSection">
          <h1 className="homeTitle">
            <span className="homeTitleText">perf.html</span>
            <span className="homeTitleSubtext">
              {' '}&mdash; Web app for Firefox performance analysis
            </span>
            <a
              className="homeTitleGithubIcon"
              href="https://github.com/devtools-html/perf.html"
            >
              <svg
                width="22"
                height="22"
                className="octicon octicon-mark-github"
                viewBox="0 0 16 16"
                version="1.1"
                aria-hidden="true"
              >
                <path
                  fillRule="evenodd"
                  d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0 0 16 8c0-4.42-3.58-8-8-8z"
                />
              </svg>
            </a>
          </h1>
          {specialMessage
            ? <div className="homeSpecialMessage">
                {specialMessage}
              </div>
            : null}
          <p>
            Capture a performance profile. Analyze it. Share it. Make the web
            faster.
          </p>
          <TransitionGroup className="homeInstructionsTransitionGroup">
            {this._renderInstructions()}
          </TransitionGroup>
          <div
            className={classNames('homeDrop', isDragging ? 'dragging' : false)}
          >
            <div className="homeDropMessage">Drop a saved profile here</div>
          </div>
        </section>
        <FooterLinks />
      </div>
    );
  }
}

function _preventDefault(event) {
  event.preventDefault();
}

function _supportsWebExtensionAPI(): boolean {
  const matched = navigator.userAgent.match(/Firefox\/(\d+\.\d+)/);
  const minimumSupportedFirefox = 55;
  return matched ? parseFloat(matched[1]) >= minimumSupportedFirefox : false;
}

function _isFirefox(): boolean {
  return Boolean(navigator.userAgent.match(/Firefox\/\d+\.\d+/));
}

export default connect(state => state, { retrieveProfileFromFile })(Home);
