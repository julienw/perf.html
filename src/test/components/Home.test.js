/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

// @flow
import * as React from 'react';
import Home from '../../components/app/Home';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import createStore from '../../create-store';

// Provide a mechanism to overwrite the navigator.userAgent, which can't be set.
const FIREFOX_WEBEXT =
  'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.12; rv:55.0) Gecko/20100101 Firefox/55.0';
const FIREFOX_LEGACY =
  'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.12; rv:53.0) Gecko/20100101 Firefox/53.0';
const SAFARI =
  'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/603.3.8 (KHTML, like Gecko) Version/10.1.2 Safari/603.3.8';
let userAgent;

// Flow doesn't understand Object.defineProperty. Use the "any" type to use it anyway.
(Object.defineProperty: any)(window.navigator, 'userAgent', {
  get: () => userAgent,
});

describe('app/Home', function() {
  it('renders the install screen for the extension', () => {
    userAgent = FIREFOX_WEBEXT;

    const home = renderer.create(
      <Provider store={createStore()}>
        <Home specialMessage="This is a special message" />
      </Provider>
    );

    expect(home.toJSON()).toMatchSnapshot();
  });

  it('renders the install screen for the legacy add-on', () => {
    userAgent = FIREFOX_LEGACY;

    const home = renderer.create(
      <Provider store={createStore()}>
        <Home specialMessage="This is a special message" />
      </Provider>
    );

    expect(home.toJSON()).toMatchSnapshot();
  });

  it('renders the information screen for other browsers', () => {
    userAgent = SAFARI;

    const home = renderer.create(
      <Provider store={createStore()}>
        <Home specialMessage="This is a special message" />
      </Provider>
    );

    expect(home.toJSON()).toMatchSnapshot();
  });

  it('renders the usage instructions for pages with the extension installed', () => {
    userAgent = FIREFOX_WEBEXT;

    window.isGeckoProfilerAddonInstalled = true;

    const home = renderer.create(
      <Provider store={createStore()}>
        <Home specialMessage="This is a special message" />
      </Provider>
    );

    expect(home.toJSON()).toMatchSnapshot();

    delete window.isGeckoProfilerAddonInstalled;
  });
});
