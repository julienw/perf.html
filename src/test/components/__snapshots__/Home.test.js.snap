// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/Home renders a button to enable the popup when it is available 1`] = `
<div
  class="homeInstructions"
  data-testid="home-enable-popup-instructions"
>
  <img
    alt="screenshot of profiler.firefox.com"
    class="homeSectionScreenshot"
    src="test-file-stub"
  />
  <div>
    <button
      class="homeSectionButton"
      type="button"
    >
      <span
        class="homeSectionPlus"
      >
        +
      </span>
      Enable Profiler Menu Button
    </button>
    <a
      class="homeSectionButton"
      href="/docs/"
    >
      <span
        class="homeSectionDocsIcon"
      />
      Documentation
    </a>
    <p>
      Enable the profiler menu button to start recording a performance profile in Firefox, then analyze it and share it with profiler.firefox.com.
    </p>
  </div>
</div>
`;

exports[`app/Home renders the information screen for other browsers 1`] = `
<div
  class="home"
>
  <main
    class="homeSection"
  >
    <header>
      <h1
        class="appHeader"
      >
        <span
          class="appHeaderSlogan"
        >
          <a
            class="appHeaderLink"
            href="/"
          >
            Firefox Profiler
          </a>
          <span
            class="appHeaderSubtext"
          >
             
            — Web app for Firefox performance analysis
          </span>
        </span>
        <a
          class="appHeaderGithubIcon"
          href="https://github.com/firefox-devtools/profiler"
          rel="noopener noreferrer"
          target="_blank"
          title="Go to our git repository (this opens in a new window)"
        >
          <svg
            aria-label="github"
            class="octicon octicon-mark-github"
            height="22"
            version="1.1"
            viewBox="0 0 16 16"
            width="22"
          >
            <path
              d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0 0 16 8c0-4.42-3.58-8-8-8z"
              fill-rule="evenodd"
            />
          </svg>
        </a>
      </h1>
    </header>
    <div
      class="homeSpecialMessage"
    >
      This is a special message
    </div>
    <p>
      Capture a performance profile. Analyze it. Share it. Make the web faster.
    </p>
    <div
      class="homeInstructionsTransitionGroup"
    >
      <div
        class="homeInstructions"
        data-testid="home-other-browser-instructions"
      >
        <img
          alt="screenshot of profiler.firefox.com"
          class="homeSectionScreenshot"
          src="test-file-stub"
        />
        <div>
          <a
            class="homeSectionButton"
            href="/docs/"
          >
            <span
              class="homeSectionDocsIcon"
            />
            Documentation
          </a>
          <h2>
            How to view and record profiles
          </h2>
          <p>
            Recording performance profiles requires
             
            <a
              href="https://www.mozilla.org/en-US/firefox/new/"
            >
              Firefox
            </a>
            . However, existing profiles can be viewed in any modern browser.
          </p>
        </div>
      </div>
    </div>
    <section
      class="homeAdditionalContent"
    >
      <h2
        class="homeAdditionalContentTitle protocol-display-xs"
      >
        Load existing profiles
      </h2>
      <section
        class="homeActions"
      >
        <p>
          You can 
          <strong>
            drag and drop
          </strong>
           a profile file here to load it, or:
        </p>
        <div
          class="homeSectionLoadProfile"
        >
          <div
            class="homeSectionActionButtons"
          >
            <label
              class="homeSectionButton"
            >
              <input
                class="homeSectionUploadFromFileInput"
                type="file"
              />
              Load a profile from file
            </label>
            <button
              aria-expanded="false"
              class="homeSectionButton"
              type="button"
            >
              Load a profile from a URL
            </button>
          </div>
        </div>
        <p>
          You can also compare recordings.
           
          <a
            href="/compare/"
          >
            Open the comparing interface.
          </a>
        </p>
      </section>
      <section>
        <h2
          class="homeRecentUploadedRecordingsTitle protocol-display-xxs"
        >
          Recent uploaded recordings
        </h2>
        <list-of-published-profiles
          limit="3"
          withactionbuttons="false"
        />
      </section>
    </section>
    <div
      class="dragAndDropOverlayWrapper"
    >
      <div
        class="dragAndDropOverlay"
      >
        Drop a saved profile here
      </div>
    </div>
  </main>
</div>
`;

exports[`app/Home renders the install screen for the extension 1`] = `
<div
  class="home"
>
  <main
    class="homeSection"
  >
    <header>
      <h1
        class="appHeader"
      >
        <span
          class="appHeaderSlogan"
        >
          <a
            class="appHeaderLink"
            href="/"
          >
            Firefox Profiler
          </a>
          <span
            class="appHeaderSubtext"
          >
             
            — Web app for Firefox performance analysis
          </span>
        </span>
        <a
          class="appHeaderGithubIcon"
          href="https://github.com/firefox-devtools/profiler"
          rel="noopener noreferrer"
          target="_blank"
          title="Go to our git repository (this opens in a new window)"
        >
          <svg
            aria-label="github"
            class="octicon octicon-mark-github"
            height="22"
            version="1.1"
            viewBox="0 0 16 16"
            width="22"
          >
            <path
              d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0 0 16 8c0-4.42-3.58-8-8-8z"
              fill-rule="evenodd"
            />
          </svg>
        </a>
      </h1>
    </header>
    <div
      class="homeSpecialMessage"
    >
      This is a special message
    </div>
    <p>
      Capture a performance profile. Analyze it. Share it. Make the web faster.
    </p>
    <div
      class="homeInstructionsTransitionGroup"
    >
      <div
        class="homeInstructions"
        data-testid="home-install-addon-instructions"
      >
        <img
          alt="screenshot of profiler.firefox.com"
          class="homeSectionScreenshot"
          src="test-file-stub"
        />
        <div>
          <a
            class="homeSectionButton"
            href="https://raw.githubusercontent.com/firefox-devtools/Gecko-Profiler-Addon/master/gecko_profiler.xpi"
          >
            <span
              class="homeSectionPlus"
            >
              +
            </span>
            Install add-on
          </a>
          <a
            class="homeSectionButton"
            href="/docs/"
          >
            <span
              class="homeSectionDocsIcon"
            />
            Documentation
          </a>
          <p>
            Install the Gecko Profiler Add-on to start recording a performance profile in Firefox, then analyze it and share it with profiler.firefox.com.
          </p>
        </div>
      </div>
    </div>
    <section
      class="homeAdditionalContent"
    >
      <h2
        class="homeAdditionalContentTitle protocol-display-xs"
      >
        Load existing profiles
      </h2>
      <section
        class="homeActions"
      >
        <p>
          You can 
          <strong>
            drag and drop
          </strong>
           a profile file here to load it, or:
        </p>
        <div
          class="homeSectionLoadProfile"
        >
          <div
            class="homeSectionActionButtons"
          >
            <label
              class="homeSectionButton"
            >
              <input
                class="homeSectionUploadFromFileInput"
                type="file"
              />
              Load a profile from file
            </label>
            <button
              aria-expanded="false"
              class="homeSectionButton"
              type="button"
            >
              Load a profile from a URL
            </button>
          </div>
        </div>
        <p>
          You can also compare recordings.
           
          <a
            href="/compare/"
          >
            Open the comparing interface.
          </a>
        </p>
      </section>
      <section>
        <h2
          class="homeRecentUploadedRecordingsTitle protocol-display-xxs"
        >
          Recent uploaded recordings
        </h2>
        <list-of-published-profiles
          limit="3"
          withactionbuttons="false"
        />
      </section>
    </section>
    <div
      class="dragAndDropOverlayWrapper"
    >
      <div
        class="dragAndDropOverlay"
      >
        Drop a saved profile here
      </div>
    </div>
  </main>
</div>
`;

exports[`app/Home renders the usage instructions for pages with the extension installed 1`] = `
<div
  class="home"
>
  <main
    class="homeSection"
  >
    <header>
      <h1
        class="appHeader"
      >
        <span
          class="appHeaderSlogan"
        >
          <a
            class="appHeaderLink"
            href="/"
          >
            Firefox Profiler
          </a>
          <span
            class="appHeaderSubtext"
          >
             
            — Web app for Firefox performance analysis
          </span>
        </span>
        <a
          class="appHeaderGithubIcon"
          href="https://github.com/firefox-devtools/profiler"
          rel="noopener noreferrer"
          target="_blank"
          title="Go to our git repository (this opens in a new window)"
        >
          <svg
            aria-label="github"
            class="octicon octicon-mark-github"
            height="22"
            version="1.1"
            viewBox="0 0 16 16"
            width="22"
          >
            <path
              d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0 0 16 8c0-4.42-3.58-8-8-8z"
              fill-rule="evenodd"
            />
          </svg>
        </a>
      </h1>
    </header>
    <div
      class="homeSpecialMessage"
    >
      This is a special message
    </div>
    <p>
      Capture a performance profile. Analyze it. Share it. Make the web faster.
    </p>
    <div
      class="homeInstructionsTransitionGroup"
    >
      <div
        class="homeInstructions"
        data-testid="home-record-instructions"
      >
        <img
          alt="Screenshot of the profiler settings from the Firefox menu."
          class="homeSectionScreenshot"
          src="test-file-stub"
        />
        <div>
          <a
            class="homeSectionButton"
            href="/docs/"
          >
            <span
              class="homeSectionDocsIcon"
            />
            Documentation
          </a>
          <p>
            To start profiling, click on the profiling button, or use the keyboard shortcuts. The icon is blue when a profile is recording. Hit
            <kbd>
              Capture Profile
            </kbd>
             to load the data into profiler.firefox.com.
          </p>
          <div>
            <p>
              <kbd>
                Ctrl
              </kbd>
              +
              <kbd>
                Shift
              </kbd>
              +
              <kbd>
                1
              </kbd>
               Stop and start profiling
            </p>
            <p>
              <kbd>
                Ctrl
              </kbd>
              +
              <kbd>
                Shift
              </kbd>
              +
              <kbd>
                2
              </kbd>
               Capture and load profile
            </p>
          </div>
        </div>
      </div>
    </div>
    <section
      class="homeAdditionalContent"
    >
      <h2
        class="homeAdditionalContentTitle protocol-display-xs"
      >
        Load existing profiles
      </h2>
      <section
        class="homeActions"
      >
        <p>
          You can 
          <strong>
            drag and drop
          </strong>
           a profile file here to load it, or:
        </p>
        <div
          class="homeSectionLoadProfile"
        >
          <div
            class="homeSectionActionButtons"
          >
            <label
              class="homeSectionButton"
            >
              <input
                class="homeSectionUploadFromFileInput"
                type="file"
              />
              Load a profile from file
            </label>
            <button
              aria-expanded="false"
              class="homeSectionButton"
              type="button"
            >
              Load a profile from a URL
            </button>
          </div>
        </div>
        <p>
          You can also compare recordings.
           
          <a
            href="/compare/"
          >
            Open the comparing interface.
          </a>
        </p>
      </section>
      <section>
        <h2
          class="homeRecentUploadedRecordingsTitle protocol-display-xxs"
        >
          Recent uploaded recordings
        </h2>
        <list-of-published-profiles
          limit="3"
          withactionbuttons="false"
        />
      </section>
    </section>
    <div
      class="dragAndDropOverlayWrapper"
    >
      <div
        class="dragAndDropOverlay"
      >
        Drop a saved profile here
      </div>
    </div>
  </main>
</div>
`;

exports[`app/Home renders the usage instructions for when the popup is enabled 1`] = `
<div
  class="homeInstructions homeTransition-enter homeTransition-enter-active"
  data-testid="home-record-instructions"
>
  <img
    alt="Screenshot of the profiler settings from the Firefox menu."
    class="homeSectionScreenshot"
    src="test-file-stub"
  />
  <div>
    <a
      class="homeSectionButton"
      href="/docs/"
    >
      <span
        class="homeSectionDocsIcon"
      />
      Documentation
    </a>
    <p>
      To start profiling, click on the profiling button, or use the keyboard shortcuts. The icon is blue when a profile is recording. Hit
      <kbd>
        Capture Profile
      </kbd>
       to load the data into profiler.firefox.com.
    </p>
    <div>
      <p>
        <kbd>
          Ctrl
        </kbd>
        +
        <kbd>
          Shift
        </kbd>
        +
        <kbd>
          1
        </kbd>
         Stop and start profiling
      </p>
      <p>
        <kbd>
          Ctrl
        </kbd>
        +
        <kbd>
          Shift
        </kbd>
        +
        <kbd>
          2
        </kbd>
         Capture and load profile
      </p>
    </div>
  </div>
</div>
`;
